### PR TITLE
Knox enhancements

### PR DESCRIPTION
**Pull request by sjsrey**
## Features

- More efficient Global Knox
- Analytical inference for Global Knox
- Local Knox
- Analytical inference for Local Knox
- Conditional permutation inference for Local Knox
- static plotting for significant clusters
- interactive plotting for significant clusters

## Todo
- ~add tests~
- ~polish user notebook (add plotting)~

### CODE REVIEW

**Review by knaaptime — APPROVED** [review]

File: pointpats/spacetime.py [hunk 1]
```diff
@@ -5,14 +5,20 @@
 __author__ = "Nicholas Malizia <nmalizia@asu.edu>", "Sergio J. Rey \
 <srey@asu.edu>", "Philip Stephens <philip.stephens@asu.edu"
 
-__all__ = ['SpaceTimeEvents', 'knox', 'mantel', 'jacquez', 'modified_knox']
+__all__ = ['SpaceTimeEvents', 'knox', 'mantel', 'jacquez', 'modified_knox',
+           'Knox', 'KnoxLocal]
```
Comment (martinfleis) [inline pointpats/spacetime.py:9]:
```suggestion
           'Knox', 'KnoxLocal']
```
this causes import error now

**Review by martinfleis — COMMENTED** [review]
This is nice! Could we try to get this and #105 in and cut a release with sklearn 1.3.0 compatibility fixes we did here?

**Comment by sjsrey** [comment]
> This is nice! Could we try to get this and #105 in and cut a release with sklearn 1.3.0 compatibility fixes we did here?

Yes. Perhaps during the sprint on 8-24?

**Review by martinfleis — APPROVED** [review]
Looking good!

**Comment by knaaptime** [comment]
@martinfleis not to distract you from working on lib at the moment, but a quick thing in case you have an idea..

we cooked up a little explore method on the localknox class that works pretty nicely. One killer thing would be to have an onclick or mouseover function that highlights neighbors when you interact with focal. I think the current method has all the info necessary to get there, but not sure how to go about rigging that up with folium. Curious if you know how

**Comment by martinfleis** [comment]
Not possible afaik. The styling function sees only a single item of the json and I didn't figure out a way of highlighting from B based on an interaction with a geom A.

**Comment by knaaptime** [comment]
bummer. pretty sure that's dead simple in vanilla leaflet. I was trying to send back IDs of neighbors as an attribute of focal to highlight those vals in a different color or something

thanks for the insight

**Comment by martinfleis** [comment]
I am not so sure, folium is pretty close in API to leaflet. But I suppose it would need a bit of custom JS which is easy when you have a leaflet map, not so much with folium. 

Maybe bokeh or geoviews may offer solutions.

**Pull request merged**